clientv3/retry.go

```diff
@@ -27,12 +27,30 @@ type retryRpcFunc func(context.Context, rpcFunc) error
 
 func (c *Client) newRetryWrapper() retryRpcFunc {
 	return func(rpcCtx context.Context, f rpcFunc) error {
+		nrAuthRetries := 0
+
 		for {
 			err := f(rpcCtx)
 			if err == nil {
 				return nil
 			}
 
+			if grpc.Code(err) == codes.Unauthenticated {
```

Comment:
probably `== ErrInvalidAuthToken`? I think this can get rid of the nrAuthRetries stuff too; if the token keeps expiring, I don't see why it shouldn't keep requesting new tokens so long as the context is still active.